pinot-spi/src/main/java/org/apache/pinot/spi/ingestion/segment/writer/SegmentWriter.java

```diff
@@ -65,6 +68,27 @@ default void collect(GenericRow[] rowBatch)
     }
   }
 
+  /**
+   * Sets the staging directory that stores the written segment files.
+   * @param tableConfig
+   */
+  default File setStagingDir(TableConfig tableConfig) {
```

Comment:
Let's move these methods into `FileBasedSegmentWriter`. I don't think they belong to the interface